lua/ttt2/libraries/vskin.lua

```diff
@@ -0,0 +1,248 @@
+---
+-- A handler of the skin colors
+-- @author Mineotopia
+
+AddCSLuaFile()
+
+-- the rest of the draw library is client only
+if SERVER then return end
+
+local cv_selectedVSkin = CreateConVar("ttt2_selected_vskin", "dark_ttt2", {FCVAR_ARCHIVE})
+local cv_blurVSkin = CreateConVar("ttt2_vskin_blur", 1, {FCVAR_ARCHIVE})
+
+vskin = vskin or {}
+
+vskin.skins = vskin.skins or {}
+
+vskin.selected = vskin.selected or ""
+
+---
+-- Register a new vskin
+-- @param string name The unique name of your vskin
+-- @param table skin The skin table
+-- @realm client
+function vskin.RegisterVSkin(name, skin)
+	if vskin.skins[name] then
+		print("[TTT2 Skin] A skin with this name already exists!")
+
+		return
+	end
+
+	vskin.skins[name] = skin
+end
+
+---
+-- Select a registered vskin
+-- @param string name The unique name of the vskin, if nil the convar value is read
+-- @return boolean Returns true if skin was selected
+-- @realm client
+function vskin.SelectVSkin(skinName)
+	if not skinName then
+		vskin.selected = cv_selectedVSkin:GetString()
+
+		return true
+	end
+
+	if not vskin.skins[skinName] then
+		return false
+	end
+
+	vskin.selected = skinName
+	cv_selectedVSkin:SetString(skinName)
```

Comment:
Why is this information saved in both the .selected and the cvar? There should only be one source of truth, otherwise this can easily lead to mistakes